src/vs/base/parts/contextmenu/electron-main/contextmenu.ts

```diff
@@ -4,22 +4,30 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { Menu, MenuItem, BrowserWindow, Event, ipcMain } from 'electron';
-import { ISerializableContextMenuItem, CONTEXT_MENU_CLOSE_CHANNEL, CONTEXT_MENU_CHANNEL, IPopupOptions } from 'vs/base/parts/contextmenu/common/contextmenu';
+import { ISerializableContextMenuItem, CONTEXT_MENU_CLOSE_CHANNEL, CONTEXT_MENU_CHANNEL, IPopupOptions, CONTEXT_MENU_HIDE_CHANNEL } from 'vs/base/parts/contextmenu/common/contextmenu';
 
 export function registerContextMenuListener(): void {
+	let menu: Menu;
 	ipcMain.on(CONTEXT_MENU_CHANNEL, (event: Event, contextMenuId: number, items: ISerializableContextMenuItem[], onClickChannel: string, options?: IPopupOptions) => {
-		const menu = createMenu(event, onClickChannel, items);
+		menu = createMenu(event, onClickChannel, items);
 
 		menu.popup({
 			window: BrowserWindow.fromWebContents(event.sender),
 			x: options ? options.x : void 0,
 			y: options ? options.y : void 0,
 			positioningItem: options ? options.positioningItem : void 0,
 			callback: () => {
+				menu = null;
 				event.sender.send(CONTEXT_MENU_CLOSE_CHANNEL, contextMenuId);
 			}
 		});
 	});
+
+	ipcMain.on(CONTEXT_MENU_HIDE_CHANNEL, (event: Event, contextMenuId: number) => {
```

Comment:
`contextMenuId` is unused?

Reply:
oops, yes, currently with this approach, the hideContextMenu call will hide any context menu that is shown, even if someone else drew it. This was experimenting with adding an id so we knew we were dismissing the right menu. it was too much code change for probably not a lot of value yet

Reply:
@sbatten ok then I would remove the variable